generic_tracers/generic_COBALT.F90

```diff
@@ -2694,58 +2696,65 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
     ! captures mixing on time scale of 1 to a few days.
     ! de Boyer-Montegut reference:  https://doi.org/10.1029/2004JC002378
     !
-    do j = jsc, jec ; do i = isc, iec   !{
+    if (present(photo_acc_dpth)) then
+      call mpp_error(WARNING, "Using a photoacclimation MLD in COBALTv3 that was calculated in "//&
```

Comment:
@yichengt900 
It would be great if there was a check here to see if photo_acc_dpth is a fixed value, but I am not sure how to do that. Any ideas?

Reply:
@theresa-morrison , do you mean to check if `photo_acc_dpth` does not contain any NaN values?

Reply:
Hi Theresa, YC - Theresa and I discussed this a bit last Friday.  In my mind, using a constant photoacclimation depth in COBALT is effectively an error.  Theresa mentioned that she wanted to keep the option to not calculate it for those folks that are using other BGC modules (or none), which seems like a good idea, but you shouldn't use COBALT without a time varying photoacclimation depth, so like the idea of passing zeros and providing an error message so the user knows there is obviously something wrong with the model and has an idea on how to solve.

Reply:
@theresa-morrison, this should be relatively straightforward to implement. If `photo_acc_dpth` is a constant array, the value at any position should be the same across the array. We can try the following trick:
```
logical :: is_all_same
is_all_same = all(photo_acc_dpth == photo_acc_dpth(isc,jsc))
if (is_all_same) then
 ! Include your error message here for COBALTv3
endif
```
If you still encounter any issues while coding this, I'm more than happy to help.

Reply:
thanks! `all` is new to me and exactly what I was looking for. I will check that this works and add it to the PR

Reply:
This has been added and works as expected.

Reply:
I should add, contrary to our expectations, COBALT does not crash with a photoacclimation MLD of 0.0 in a shourt 24 hour run.